### ISSUE TITLE
task: set up conda-forge packaging

### ISSUE THREAD

**Issue by alisterburt** [warpem/warp#16]
issue to track ongoing packaging effort at https://github.com/conda-forge/staged-recipes/pull/25492

- [x] depends on #15 
- [ ] get recipe in passing state in staged-recipes
- [ ] get recipe merged and first version distributed
- [ ] automate the updating of the feedstock

non goals of packaging effort as of right now:
- windows or osx builds (although a [GPU runner for windows builds](https://conda-forge.org/docs/maintainer/knowledge_base/#packages-that-require-a-gpu-or-long-running-builds) might be possible)

**Comment by alisterburt** [comment]
pipeline trying to build against current version of main d9b13c2

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=920573&view=results

**Comment by alisterburt** [comment]
That build is failing.

When I try to build an environment locally with the conda_build.yml in the repo at d9b13c2

```yaml
name: warp_build
channels:
  - conda-forge
  - nodefaults
dependencies:
  - torchvision
  - libtiff
  - python
  - pytorch=1.11
  - pytorch-gpu
  - fftw
  - cmake
  - cxx-compiler=1.3
  - dotnet=8.0
  - cudatoolkit
  - cudatoolkit-dev
  - cudnn
```

It resolves as follows:
```txt
# packages in environment at /home/burta2/mambaforge/envs/warp_build:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                  2_kmp_llvm    conda-forge
binutils                  2.39                 hdd6e379_1    conda-forge
binutils_impl_linux-64    2.39                 he00db2b_1    conda-forge
binutils_linux-64         2.39                h5fc0e48_13    conda-forge
brotli-python             1.1.0           py310hc6cd4ac_1    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
c-ares                    1.28.1               hd590300_0    conda-forge
c-compiler                1.3.0                h7f98852_0    conda-forge
ca-certificates           2024.2.2             hbcca054_0    conda-forge
certifi                   2024.2.2           pyhd8ed1ab_0    conda-forge
cffi                      1.16.0          py310h2fee648_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
cmake                     3.29.2               hcfe8598_0    conda-forge
cuda-version              11.8                 h70ddcb2_3    conda-forge
cudatoolkit               11.8.0              h4ba93d1_13    conda-forge
cudatoolkit-dev           11.7.0               h1de0b5d_6    conda-forge
cudnn                     8.9.7.29             hbc23b4c_3    conda-forge
cxx-compiler              1.3.0                h4bd325d_0    conda-forge
dotnet                    8.0.204              ha770c72_0    conda-forge
dotnet-aspnetcore         8.0.4                hb8a3ed7_0    conda-forge
dotnet-runtime            8.0.4                hb8a3ed7_0    conda-forge
dotnet-sdk                8.0.204              hb8a3ed7_0    conda-forge
fftw                      3.3.10          nompi_hc118613_108    conda-forge
freetype                  2.12.1               h267a509_2    conda-forge
gcc                       9.5.0               h1fea6ba_13    conda-forge
gcc_impl_linux-64         9.5.0               h99780fb_19    conda-forge
gcc_linux-64              9.5.0               h4258300_13    conda-forge
gxx                       9.5.0               h1fea6ba_13    conda-forge
gxx_impl_linux-64         9.5.0               h99780fb_19    conda-forge
gxx_linux-64              9.5.0               h43f449f_13    conda-forge
icu                       73.2                 h59595ed_0    conda-forge
idna                      3.7                pyhd8ed1ab_0    conda-forge
kernel-headers_linux-64   2.6.32              he073ed8_17    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
lcms2                     2.16                 hb7c19ff_0    conda-forge
ld_impl_linux-64          2.39                 hcc3a1bd_1    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
libblas                   3.9.0            16_linux64_mkl    conda-forge
libcblas                  3.9.0            16_linux64_mkl    conda-forge
libcurl                   8.7.1                hca28451_0    conda-forge
libdeflate                1.20                 hd590300_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 hd590300_2    conda-forge
libexpat                  2.6.2                h59595ed_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-devel_linux-64     9.5.0               h0a57e50_19    conda-forge
libgcc-ng                 13.2.0               hc881cc4_6    conda-forge
libgfortran-ng            13.2.0               h69a702a_6    conda-forge
libgfortran5              13.2.0               h43f5ff8_6    conda-forge
libgomp                   13.2.0               hc881cc4_6    conda-forge
libhwloc                  2.10.0          default_h2fb2949_1000    conda-forge
libiconv                  1.17                 hd590300_2    conda-forge
libjpeg-turbo             3.0.0                hd590300_1    conda-forge
liblapack                 3.9.0            16_linux64_mkl    conda-forge
libnghttp2                1.58.0               h47da74e_1    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libpng                    1.6.43               h2797004_0    conda-forge
libprotobuf               3.20.3               h3eb15da_0    conda-forge
libsanitizer              9.5.0               h2f262e1_19    conda-forge
libsqlite                 3.45.3               h2797004_0    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-devel_linux-64  9.5.0               h0a57e50_19    conda-forge
libstdcxx-ng              13.2.0               h95c4c6d_6    conda-forge
libtiff                   4.6.0                h1dd3fc0_3    conda-forge
liburcu                   0.13.2               h166bdaf_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libuv                     1.48.0               hd590300_0    conda-forge
libwebp-base              1.4.0                hd590300_0    conda-forge
libxcb                    1.15                 h0b41bf4_0    conda-forge
libxcrypt                 4.4.36               hd590300_1    conda-forge
libxml2                   2.12.6               h232c23b_2    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
llvm-openmp               18.1.3               h4dfa4b3_0    conda-forge
lttng-ust                 2.13.4               hfdfcbd3_0    conda-forge
magma                     2.5.4                hc72dce7_4    conda-forge
mkl                       2022.2.1         h84fe81f_16997    conda-forge
nccl                      2.21.5.1             h6103f9b_0    conda-forge
ncurses                   6.4.20240210         h59595ed_0    conda-forge
ninja                     1.12.0               h00ab1b0_0    conda-forge
numpy                     1.26.4          py310hb13e2d6_0    conda-forge
openjpeg                  2.5.2                h488ebb8_0    conda-forge
openssl                   3.2.1                hd590300_1    conda-forge
pillow                    10.3.0          py310hf73ecf8_0    conda-forge
pip                       24.0               pyhd8ed1ab_0    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
pycparser                 2.22               pyhd8ed1ab_0    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.10.14         hd12c33a_0_cpython    conda-forge
python_abi                3.10                    4_cp310    conda-forge
pytorch                   1.11.0          cuda112py310h51fe464_1    conda-forge
pytorch-gpu               1.11.0          cuda112py310h68407e5_1    conda-forge
readline                  8.2                  h8228510_1    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
rhash                     1.4.4                hd590300_0    conda-forge
setuptools                69.5.1             pyhd8ed1ab_0    conda-forge
sleef                     3.5.1                h9b69904_2    conda-forge
sysroot_linux-64          2.12                he073ed8_17    conda-forge
tbb                       2021.12.0            h00ab1b0_0    conda-forge
tk                        8.6.13          noxft_h4845f30_101    conda-forge
torchvision               0.15.2          cuda112py310h993b30e_2    conda-forge
typing_extensions         4.11.0             pyha770c72_0    conda-forge
tzdata                    2024a                h0c530f3_0    conda-forge
urllib3                   2.2.1              pyhd8ed1ab_0    conda-forge
wheel                     0.43.0             pyhd8ed1ab_1    conda-forge
xorg-libxau               1.0.11               hd590300_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zstd                      1.5.5                hfc55251_0    conda-forge
```

It seems like torchvision is [the sticking point](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=920607&view=logs&j=eb6ecf2b-0363-5c1a-baa7-2d771889d197&t=248aa990-2a52-51cf-fcc8-1e0aefce9a33&l=1305) in the conda-forge build. 

```txt
torchvision is not installable because there are no viable options
```

specifically 0.15.2 complains because "[torchvision 0.15.2 would require](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=920607&view=logs&j=eb6ecf2b-0363-5c1a-baa7-2d771889d197&t=248aa990-2a52-51cf-fcc8-1e0aefce9a33&l=1305)"

Ultimately it complains torchvision==0.15.2 is [incompatible with pytorch==1.11.0](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=920607&view=logs&j=eb6ecf2b-0363-5c1a-baa7-2d771889d197&t=248aa990-2a52-51cf-fcc8-1e0aefce9a33&l=1308)

This is because PyTorch 1.11 wants these things:
```txt
2024-04-23T14:33:11.6798336Z ├─ pytorch 1.11.* cuda* is installable with the potential options
2024-04-23T14:33:11.6799034Z │  ├─ pytorch [1.10.0|1.10.1|...|1.9.1] would require
2024-04-23T14:33:11.6799491Z │  │  └─ cudnn [>=7.6,<8.0a0 |>=7.6.5.32,<8.0a0 ], which conflicts with any installable versions previously reported;
2024-04-23T14:33:11.6799820Z │  ├─ pytorch 1.11.0 would require
2024-04-23T14:33:11.6800152Z │  │  ├─ cudnn >=8.2.1.32,<9.0a0  with the potential options
2024-04-23T14:33:11.6800522Z │  │  │  ├─ cudnn [8.2.1.32|8.3.2.44|8.4.0.27], which can be installed;
2024-04-23T14:33:11.6801048Z │  │  │  ├─ cudnn 8.4.1.50, which can be installed;
2024-04-23T14:33:11.6848026Z │  │  │  ├─ cudnn 8.8.0.121, which can be installed;
2024-04-23T14:33:11.6851568Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which cannot be installed (as previously explained);
2024-04-23T14:33:11.6854728Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which can be installed (as previously explained);
2024-04-23T14:33:11.6857825Z │  │  │  ├─ cudnn 8.9.7.29, which can be installed;
2024-04-23T14:33:11.6860932Z │  │  │  └─ cudnn 8.9.7.29, which can be installed (as previously explained);
2024-04-23T14:33:11.6865129Z │  │  └─ python_abi 3.10.* *_cp310 but there are no viable options
2024-04-23T14:33:11.6868455Z │  │     ├─ python_abi 3.10 conflicts with any installable versions previously reported;
2024-04-23T14:33:11.6871907Z │  │     └─ python_abi 3.10 would require
2024-04-23T14:33:11.6875320Z │  │        └─ python 3.10.* *_cpython, which conflicts with any installable versions previously reported;
2024-04-23T14:33:11.6878325Z │  ├─ pytorch 1.11.0 would require
2024-04-23T14:33:11.6881337Z │  │  ├─ cudnn >=8.2.1.32,<9.0a0  with the potential options
2024-04-23T14:33:11.6884354Z │  │  │  ├─ cudnn [8.2.1.32|8.3.2.44|8.4.0.27], which can be installed;
2024-04-23T14:33:11.6887318Z │  │  │  ├─ cudnn 8.4.1.50, which can be installed;
2024-04-23T14:33:11.6890326Z │  │  │  ├─ cudnn 8.8.0.121, which can be installed;
2024-04-23T14:33:11.6893691Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which cannot be installed (as previously explained);
2024-04-23T14:33:11.6896752Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which can be installed (as previously explained);
2024-04-23T14:33:11.6901291Z │  │  │  ├─ cudnn 8.9.7.29, which can be installed;
2024-04-23T14:33:11.6904943Z │  │  │  └─ cudnn 8.9.7.29, which can be installed (as previously explained);
2024-04-23T14:33:11.6908183Z │  │  └─ python_abi 3.7.* *_cp37m but there are no viable options
2024-04-23T14:33:11.6911184Z │  │     ├─ python_abi 3.7 conflicts with any installable versions previously reported;
2024-04-23T14:33:11.6914447Z │  │     └─ python_abi 3.7 would require
2024-04-23T14:33:11.6917498Z │  │        └─ python 3.7.* *_cpython, which conflicts with any installable versions previously reported;
2024-04-23T14:33:11.6920427Z │  ├─ pytorch 1.11.0 would require
2024-04-23T14:33:11.6922207Z │  │  ├─ cudnn >=8.2.1.32,<9.0a0  with the potential options
2024-04-23T14:33:11.6923039Z │  │  │  ├─ cudnn [8.2.1.32|8.3.2.44|8.4.0.27], which can be installed;
2024-04-23T14:33:11.6923851Z │  │  │  ├─ cudnn 8.4.1.50, which can be installed;
2024-04-23T14:33:11.6924586Z │  │  │  ├─ cudnn 8.8.0.121, which can be installed;
2024-04-23T14:33:11.6925360Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which cannot be installed (as previously explained);
2024-04-23T14:33:11.6926228Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which can be installed (as previously explained);
2024-04-23T14:33:11.6927091Z │  │  │  ├─ cudnn 8.9.7.29, which can be installed;
2024-04-23T14:33:11.6927835Z │  │  │  └─ cudnn 8.9.7.29, which can be installed (as previously explained);
2024-04-23T14:33:11.6928645Z │  │  └─ python_abi 3.8.* *_cp38 but there are no viable options
2024-04-23T14:33:11.6929531Z │  │     ├─ python_abi 3.8 conflicts with any installable versions previously reported;
2024-04-23T14:33:11.6930230Z │  │     └─ python_abi 3.8 would require
2024-04-23T14:33:11.6931026Z │  │        └─ python 3.8.* *_cpython, which conflicts with any installable versions previously reported;
2024-04-23T14:33:11.6932445Z │  ├─ pytorch 1.11.0 would require
2024-04-23T14:33:11.6933157Z │  │  ├─ cudnn >=8.2.1.32,<9.0a0  with the potential options
2024-04-23T14:33:11.6933982Z │  │  │  ├─ cudnn [8.2.1.32|8.3.2.44|8.4.0.27], which can be installed;
2024-04-23T14:33:11.6934890Z │  │  │  ├─ cudnn 8.4.1.50, which can be installed;
2024-04-23T14:33:11.6935599Z │  │  │  ├─ cudnn 8.8.0.121, which can be installed;
2024-04-23T14:33:11.6936395Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which cannot be installed (as previously explained);
2024-04-23T14:33:11.6937497Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which can be installed (as previously explained);
2024-04-23T14:33:11.6938368Z │  │  │  ├─ cudnn 8.9.7.29, which can be installed;
2024-04-23T14:33:11.6939123Z │  │  │  └─ cudnn 8.9.7.29, which can be installed (as previously explained);
2024-04-23T14:33:11.6939845Z │  │  └─ pytorch-cpu 99999999* , which can be installed;
2024-04-23T14:33:11.6940682Z │  └─ pytorch 1.11.0 would require
2024-04-23T14:33:11.6942429Z │     ├─ cudnn >=8.2.1.32,<9.0a0  with the potential options
2024-04-23T14:33:11.6943818Z │     │  ├─ cudnn [8.2.1.32|8.3.2.44|8.4.0.27], which can be installed;
2024-04-23T14:33:11.6944538Z │     │  ├─ cudnn 8.4.1.50, which can be installed;
2024-04-23T14:33:11.6953761Z │     │  ├─ cudnn 8.8.0.121, which can be installed;
2024-04-23T14:33:11.6954974Z │     │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which cannot be installed (as previously explained);
2024-04-23T14:33:11.6955764Z │     │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which can be installed (as previously explained);
2024-04-23T14:33:11.6956473Z │     │  ├─ cudnn 8.9.7.29, which can be installed;
2024-04-23T14:33:11.6956998Z │     │  └─ cudnn 8.9.7.29, which can be installed (as previously explained);
2024-04-23T14:33:11.6957487Z │     ├─ pytorch-cpu 99999999* , which can be installed;
2024-04-23T14:33:11.6957993Z │     └─ pytorch-gpu 1.11.0* , which can be installed;
```

whilst torchvision 0.15.2 requires these things

```txt
2024-04-23T14:33:11.7240044Z │  ├─ torchvision [0.15.1|0.15.2|0.16.1] would require
2024-04-23T14:33:11.7240503Z │  │  └─ python_abi 3.11.* *_cp311 but there are no viable options
2024-04-23T14:33:11.7241030Z │  │     ├─ python_abi 3.11 conflicts with any installable versions previously reported;
2024-04-23T14:33:11.7241681Z │  │     └─ python_abi 3.11, which cannot be installed (as previously explained);
2024-04-23T14:33:11.7242127Z │  ├─ torchvision [0.15.1|0.15.2|0.16.1] would require
2024-04-23T14:33:11.7242546Z │  │  ├─ cudnn >=8.8.0.121,<9.0a0  with the potential options
2024-04-23T14:33:11.7242977Z │  │  │  ├─ cudnn 8.8.0.121, which can be installed;
2024-04-23T14:33:11.7243445Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7244173Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which can be installed (as previously explained);
2024-04-23T14:33:11.7244640Z │  │  │  ├─ cudnn 8.9.7.29, which can be installed;
2024-04-23T14:33:11.7245083Z │  │  │  └─ cudnn 8.9.7.29, which can be installed (as previously explained);
2024-04-23T14:33:11.7245552Z │  │  └─ python_abi 3.10.* *_cp310 but there are no viable options
2024-04-23T14:33:11.7246025Z │  │     ├─ python_abi 3.10 conflicts with any installable versions previously reported;
2024-04-23T14:33:11.7246698Z │  │     └─ python_abi 3.10, which cannot be installed (as previously explained);
2024-04-23T14:33:11.7247158Z │  ├─ torchvision [0.15.1|0.15.2|0.16.1] would require
2024-04-23T14:33:11.7247582Z │  │  ├─ cudnn >=8.8.0.121,<9.0a0  with the potential options
2024-04-23T14:33:11.7248012Z │  │  │  ├─ cudnn 8.8.0.121, which can be installed;
2024-04-23T14:33:11.7248486Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7249336Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which can be installed (as previously explained);
2024-04-23T14:33:11.7249806Z │  │  │  ├─ cudnn 8.9.7.29, which can be installed;
2024-04-23T14:33:11.7250257Z │  │  │  └─ cudnn 8.9.7.29, which can be installed (as previously explained);
2024-04-23T14:33:11.7250726Z │  │  └─ python_abi 3.11.* *_cp311 but there are no viable options
2024-04-23T14:33:11.7251202Z │  │     ├─ python_abi 3.11 conflicts with any installable versions previously reported;
2024-04-23T14:33:11.7251931Z │  │     └─ python_abi 3.11, which cannot be installed (as previously explained);
2024-04-23T14:33:11.7252550Z │  ├─ torchvision [0.15.1|0.15.2|0.16.1] would require
2024-04-23T14:33:11.7253173Z │  │  ├─ cudnn >=8.8.0.121,<9.0a0  with the potential options
2024-04-23T14:33:11.7253625Z │  │  │  ├─ cudnn 8.8.0.121, which can be installed;
2024-04-23T14:33:11.7254129Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7254695Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which can be installed (as previously explained);
2024-04-23T14:33:11.7255158Z │  │  │  ├─ cudnn 8.9.7.29, which can be installed;
2024-04-23T14:33:11.7255640Z │  │  │  └─ cudnn 8.9.7.29, which can be installed (as previously explained);
2024-04-23T14:33:11.7256102Z │  │  └─ python_abi 3.8.* *_cp38 but there are no viable options
2024-04-23T14:33:11.7256771Z │  │     ├─ python_abi 3.8 conflicts with any installable versions previously reported;
2024-04-23T14:33:11.7257435Z │  │     └─ python_abi 3.8, which cannot be installed (as previously explained);
2024-04-23T14:33:11.7258027Z │  ├─ torchvision [0.15.1|0.15.2] would require
2024-04-23T14:33:11.7258662Z │  │  ├─ cudnn >=8.8.0.121,<9.0a0  with the potential options
2024-04-23T14:33:11.7259125Z │  │  │  ├─ cudnn 8.8.0.121, which can be installed;
2024-04-23T14:33:11.7259684Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7260671Z │  │  │  ├─ cudnn [8.8.0.121|8.9.1.23|8.9.7.29], which can be installed (as previously explained);
2024-04-23T14:33:11.7261201Z │  │  │  ├─ cudnn 8.9.7.29, which can be installed;
2024-04-23T14:33:11.7261858Z │  │  │  └─ cudnn 8.9.7.29, which can be installed (as previously explained);
2024-04-23T14:33:11.7262508Z │  │  └─ pytorch [2.0 cuda112*|2.0.* cuda112*], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7263264Z │  ├─ torchvision 0.15.2 would require
2024-04-23T14:33:11.7263752Z │  │  └─ pytorch >=2.0.0,<2.1.0a0  but there are no viable options
2024-04-23T14:33:11.7264509Z │  │     ├─ pytorch [1.10.0|1.10.1|...|2.1.2], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7265215Z │  │     ├─ pytorch [1.11.0|1.12.0|...|2.1.2], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7265737Z │  │     ├─ pytorch [1.13.0|1.13.1|2.0.0|2.1.0|2.1.2], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7266424Z │  │     ├─ pytorch 2.0.0 conflicts with any installable versions previously reported;
2024-04-23T14:33:11.7266916Z │  │     ├─ pytorch [2.0.0|2.1.0|2.1.2], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7267443Z │  │     ├─ pytorch [1.12.0|1.12.1|1.13.0|1.13.1|2.0.0], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7267957Z │  │     ├─ pytorch [1.13.0|1.13.1|2.0.0], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7268651Z │  │     ├─ pytorch [2.0.0|2.1.0|2.1.2], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7269419Z │  │     ├─ pytorch [1.12.0|1.12.1|1.13.0|1.13.1|2.0.0], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7269946Z │  │     ├─ pytorch [2.0.0|2.1.0|2.1.2], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7270677Z │  │     ├─ pytorch [2.0.0|2.1.0|2.1.2], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7271279Z │  │     ├─ pytorch [1.12.0|1.12.1|1.13.0|1.13.1|2.0.0], which cannot be installed (as previously explained);
2024-04-23T14:33:11.7271841Z │  │     └─ pytorch [2.0.0|2.1.0|2.1.2], which cannot be installed (as previously explained);
```

There seems to be some ABI incompatibility.

Any ideas @dtegunov ?

**Comment by alisterburt** [comment]
After discussing with @dtegunov 
1. we're blocked on conda-forge packaging as mentioned above and are looking into contracting with Quansight for support
2. we're going to look at building/deploying ourselves with github actions and our own channel

Option 1 is preferred as we can leverage their build infrastructure and get guaranteed compatibility with their standard libraries. Option 2 is our clearest path to a simple installation for our users.

**Comment by alisterburt** [comment]
Investigating option 2:

relevant conda-forge docs: https://conda.io/projects/conda/en/latest/user-guide/tasks/create-custom-channels.html

A channel seems to be a local or web hosted folder structure containing tarballs. The tarballs can be built with conda-build by you in your own environment.

I haven't yet tried to run conda-build in our infrastructure but setting it up amounts to:
- installing `conda-build`
- specifying a recipe as we have done on conda-forge
- running conda-build 

We can be more flexible with specifying our requirements in a package if we don't have to adhere to conda-forge's rules, this should work

**Comment by dtegunov** [comment]
Since we're hosting the package under our own channel now, I think we can drop the conda-forge path. Just need a GitHub pipeline to build it non-locally.

**Comment by alisterburt** [comment]
closed by #58 - not conda-forge but conda with our own warpem channel as the CUDA provided by conda-forge proved tricky to work with
